internal/exec/engine.go

```diff
@@ -53,10 +53,11 @@ type Engine struct {
 
 // Run executes the stage of the given name. It returns true if the stage
 // successfully ran and false if there were any errors.
-func (e Engine) Run(stageName string) bool {
+func (e Engine) Run(stageName string) (bool, error) {
 	if e.Fetcher == nil || e.Logger == nil {
 		fmt.Fprintf(os.Stderr, "engine incorrectly configured\n")
-		return false
+		err := errors.ErrEngineConfiguration
+		return false, err
```

Comment:
You can just directly return `errors.ErrEngineConfiguration` rather than using a temporary variable.